src/Validate.Lib/Validators/ColumnValidator.cs

```diff
@@ -0,0 +1,165 @@
+﻿
+namespace FormatValidator.Validators
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// A collection of column validators that are orchestrated over a single
+    /// row of the file being validated.
+    /// </summary>
+    /// <seealso cref="Validator"/>
+    internal class ColumnValidator
+    {
+        private ValidatorGroup[] _columns;
+        private ColumnValidationError _errorInformation;
+        private string _columnSeperator;
+
+        public ColumnValidator()
+        {
+            _errorInformation = new ColumnValidationError();
+            _columns = new ValidatorGroup[0];
+        }
+
+        public ColumnValidator(string columnSeperator) : this()
+        {
+            _columnSeperator = columnSeperator;
+        }
+
+        /// <summary>
+        /// Runs all of the column validators against the string provided in <paramref name="toCheck"/>.
+        /// </summary>
+        /// <param name="toCheck">The row of data to validate</param>
+        /// <returns>True if the row has passed validation else false.</returns>
+        public bool IsValid(string toCheck)
+        {
+            bool isValid = true;
+            string[] parts = ColumnSplitter.Split(toCheck, _columnSeperator);
+            int[] columnIndexes = CalculateColumnStartIndexes(parts);
+
+            for (int currentColumn = 0; currentColumn < parts.Length; currentColumn++)
+            {
+                if (currentColumn < _columns.Length)
+                {
+                    string elemToCheck = parts[currentColumn].Trim().Replace("\"", "");
+                    bool result = _columns[currentColumn].IsValid(elemToCheck);
+
+                    IList<ValidationError> newErrors = _columns[currentColumn].GetErrors();
+                    _errorInformation.Errors.AddRange(newErrors);
+
+                    // set validation character error location for all errors
+                    // on this column to the first character in the column
+                    for (int i = 0; i < newErrors.Count; i++)
+                    {
+                        newErrors[i].AtCharacter = columnIndexes[currentColumn];
+                        newErrors[i].Column = currentColumn + 1; // zero based index
+                    }
+
+                    isValid = isValid & result;
```

Comment:
Took this from Microsoft docs, as it is, this is correct. "The & operator computes the logical AND of its operands. The result of x & y is true if both x and y evaluate to true The & operator evaluates both operands even if the left-hand operand evaluates to false , so that the operation result is false regardless of the value of the right-hand operand."